beacon/cached_provider_test.go

```diff
@@ -0,0 +1,75 @@
+package beacon
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+)
+
+func TestBeaconCache(t *testing.T) {
+	ctrl := gomock.NewController(t)
```

Comment:
NIT: if a subtest fails this will report the wrong test to be failing (since the `testing.T` of the main test is used).